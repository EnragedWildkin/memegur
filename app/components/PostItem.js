import React, { Component } from 'react';
import { View, Text, Image } from 'react-native';
import PostSection from './PostSection';
import PostCounterWithIcon from './PostCounterWithIcon';

class PostItem extends Component {
  render() {
    return(
      // TODO: Post component - https://github.com/EnragedWildkin/memegur/issues/15
      <PostSection>
        {/* TODO: Image component - https://github.com/EnragedWildkin/memegur/issues/14 */}
        <Image
          style={styles.postImageStyle}
          source={{ uri: this.props.post.source }}
        />

        {/* TODO: Counters list component - https://github.com/EnragedWildkin/memegur/issues/13 */}
        <View style={{ flex: 1, justifyContent: 'space-around' }}>
          <PostCounterWithIcon value={this.props.post.rates} icon='arrow-round-up' />
          <PostCounterWithIcon value={this.props.post.comments} icon='chatboxes' />
          <PostCounterWithIcon value={this.props.post.views} icon='eye' />
        </View>
      </PostSection>
    );
  }
}

const styles = {
  postImageStyle: {
    flex: 3,
    height: 150,
    width: 150
  }
};

export default PostItem;
