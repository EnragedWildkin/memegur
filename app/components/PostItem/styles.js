import colors from '../../constants/Colors';

const styles = {
  postContainerStyle: {
    backgroundColor: colors.postBackground,
    padding: 10,
    paddingBottom: 0,
    margin: 10,
    marginBottom: 0,
  },
  postImageStyle: {
    flex: 4,
    height: 150,
    width: 150,
  },
  postTitleStyle: {
    padding: 5,
    paddingBottom: 10,
    color: '#fff',
    fontSize: 14,
    flex: 1,
  },
};

export default styles;
