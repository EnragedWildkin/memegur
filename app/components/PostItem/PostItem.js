import React, { Component } from 'react';
import { View, Text, Image } from 'react-native';
import PostSection from '../PostSection';
import PostCounters from '../PostCounters';
import styles from './styles';

class PostItem extends Component {
  render() {
    const { postContainerStyle, postTitleStyle, postImageStyle } = styles;

    return (
      <View style={postContainerStyle}>
        <PostSection style={{ flexDirection: 'column' }}>
          <Text
            numberOfLines={1}
            style={postTitleStyle}>
            {this.props.post.title}
          </Text>

          <View style={{ flexDirection: 'row' }}>
            {/* TODO: Image component - https://github.com/EnragedWildkin/memegur/issues/14 */}
            <Image
              style={postImageStyle}
              source={{ uri: this.props.post.source }}
            />

            <PostCounters post={this.props.post}/>
          </View>
        </PostSection>
      </View>
    );
  }
}

export default PostItem;
