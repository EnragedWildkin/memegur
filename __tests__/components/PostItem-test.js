import 'react-native';
import renderer from 'react-test-renderer';
import React from 'react';
import PostItem from '../../app/components/PostItem/PostItem';

describe('<Post Item />', () => {
  const POST = {
    title: 'Test title',
    source: 'example.com/url',
    rates: 10,
    comments: 10,
    views: 10,
  };
  
  it('renders post information', () => {
    const tree = renderer.create(
      <PostItem post={POST} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
