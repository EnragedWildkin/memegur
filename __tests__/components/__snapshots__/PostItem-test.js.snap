// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Post Item /> renders post information 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#333",
      "padding": 10,
      "paddingBottom": 0,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "dimgray",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "position": "relative",
        },
        Object {
          "flexDirection": "column",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Object {
          "color": "#fff",
          "flex": 1,
          "fontSize": 14,
          "padding": 5,
          "paddingBottom": 10,
        }
      }
    >
      Test title
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "example.com/url",
          }
        }
        style={
          Object {
            "flex": 4,
            "height": 150,
            "width": 150,
          }
        }
      />
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "space-around",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "column",
            }
          }
        >
          <Text
            accessible={true}
            active={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Object {
                  "color": "lightgray",
                  "fontSize": 18,
                },
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
            virtual={undefined}
          >
            
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "lightgray",
                "fontSize": 12,
                "fontWeight": "bold",
              }
            }
          >
            10
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "column",
            }
          }
        >
          <Text
            accessible={true}
            active={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Object {
                  "color": "lightgray",
                  "fontSize": 18,
                },
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
            virtual={undefined}
          >
            
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "lightgray",
                "fontSize": 12,
                "fontWeight": "bold",
              }
            }
          >
            10
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "column",
            }
          }
        >
          <Text
            accessible={true}
            active={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Object {
                  "color": "lightgray",
                  "fontSize": 18,
                },
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
            virtual={undefined}
          >
            
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "lightgray",
                "fontSize": 12,
                "fontWeight": "bold",
              }
            }
          >
            10
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
